beanie/odm/operators/find/logical.py

```diff
@@ -148,4 +148,19 @@ def __init__(self, expression: Mapping[str, Any]):
 
     @property
     def query(self):
-        return {"$not": self.expression}
+        if len(self.expression) == 1:
+            expression_key = list(self.expression.keys())[0]
+            if expression_key.startswith("$"):
+                raise AttributeError(
+                    "Not operator can not be used with operators"
+                )
+            value = self.expression[expression_key]
+            if isinstance(value, dict):
+                internal_key = list(value.keys())[0]
+                if internal_key.startswith("$"):
+                    return {expression_key: {"$not": value}}
```

Comment:
@roman-right is the (omitted) `else` part here even valid, e.g. `{"$not": {"foo": 3}}`? All the examples I have seen with dict value are operator expressions, i.e. the condition here is always true. If that's the case this can be simplified. The added test doesn't cover this case so I can't tell for sure.